src/Microsoft.IdentityModel.Tokens/AsymmetricSignatureProvider.cs

```diff
@@ -469,17 +471,15 @@ private void CreateECDsaFromJsonWebKey(JsonWebKey webKey, bool willCreateSignatu
                     Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
                     using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPrivateBlob))
                     {
-                        _ecdsa = new ECDsaCng(cngKey);
-                        _disposeEcdsa = true;
+                        return new ECDsaCng(cngKey);
                     }
                 }
                 else
                 {
                     Marshal.Copy(keyBlobPtr, keyBlob, 0, keyBlob.Length);
                     using (CngKey cngKey = CngKey.Import(keyBlob, CngKeyBlobFormat.EccPublicBlob))
                     {
-                        _ecdsa = new ECDsaCng(cngKey);
-                        _disposeEcdsa = true;
+                        return new ECDsaCng(cngKey);
```

Comment:
We have test cases for different key size. But like I mentioned above, we cannot set ECDsa curve manually in current API.